### PR TITLE
Add TTL to storage

### PR DESCRIPTION
**Pull request by Etroid**

### CODE REVIEW

**Comment by doublejosh** [comment]
Time for some DOC updates.  I've been meaning to move everything over to wiki pages for some time... versioning markdown files is no fun.

**Comment by doublejosh** [comment]
Looks like the ttl param being present didn't break any storage backends being tested in Grunt, woot!

* [ ] We'll need to update these included storage backends, because the param may be slightly different... https://github.com/tableau-mkt/groucho/blob/master/test/groucho_mock.js

**Comment by doublejosh** [comment]
Hate to keep cascading these changes, but it looks like you've added a required param for the configs... which means a breaking API change yes?

Also a change to the implementing module...
https://github.com/tableau-mkt/semi_anonymous/blob/7.x-2.x/semi_anonymous.module#L136

**Comment by Etroid** [comment]
@doublejosh adjusted the code so that older configs continue working as expected.

**Comment by doublejosh** [comment]
This will need the update to semi_anonymous module as well.

File: test/groucho_mock.js [hunk 1]
```diff
@@ -18,18 +18,37 @@ var groucho = window.groucho || {};
...
+          time: new Date().getTime()
+        });
       },
       get: function get(id) {
-        return store.get(id);
+        var info = store.get(id),
+            now = new Date().getTime();
+
+        if (!info || typeof(info.time) === 'undefined'
+            || typeof(info.ttl) === 'undefined') {
```
Comment (doublejosh) [inline test/groucho_mock.js]: F U hound-bot

**Comment by doublejosh** [comment]
I'm thinkin' `0.3.0` or we can go crazy and call it `1.0.0`

**Comment by doublejosh** [comment]
If we merge this one, then you don't HAVE to add the new config to Semi Anonymous... but it would be real nice to add that config option.

https://github.com/tableau-mkt/groucho/pull/25

**Comment by angrytoast** [comment]
Merging per discussion with @doublejosh

**Pull request merged**